blog/2025-03-17.md

```diff
@@ -0,0 +1,98 @@
+
+* Use case / user story.
+* Architecture diagram(s), if any additional components are needed to complement
+  the reference architecture in Docker Compose.
+* Ideally - but not required - screenshots showing a system in action, no matter
+  how rough looking it may be.
+
+Note that a live demonstration will not be possible from the front of the room
+during the Tech Congress session. Very short videos may be accepted, though we
+don't know in advance how well any audio will work.
+
+## Context
+
+Before getting to the ideas, let's review what is in milestone 0.4.0:
+
+* Two ASP.NET applications:
+  * Data Management Service (DMS):
+    * [Ed-Fi Discovery API](/reference/data-exchange/api-guidelines/design-and-implementation-guidelines/api-design-guidelines/discovery-api)
+      * Including Open API specifications.
+    * [Ed-Fi Resources API](/reference/data-exchange/api-guidelines/design-and-implementation-guidelines/api-design-guidelines/resources/)
+      * Data Standard 5.2 with TPDM 1.0.
+    * [Ed-Fi Descriptors API](/reference/data-exchange/api-guidelines/design-and-implementation-guidelines/api-design-guidelines/ed-fi-descriptors)
+  * DMS Configuration Service:
+    * Mostly compatible with the [Admin API 2.2 specification](https://github.com/Ed-Fi-Alliance-OSS/Ed-Fi-API-Standards/blob/main/api-specifications/admin-api/admin-api-2.2.0.yaml).
+* Storage:
+  * PostgreSQL for modifications
+  * OpenSearch or Elasticsearch for queries.
+* Event streaming in Kafka.
+* Authorization (see note):
+  * Client authorization (you're allowed to use the API) via Keycloak.
+  * Resource authorization (you're allowed to access this endpoint, using claim sets).
+  * Document authorization, via namespaces (you're allowed to access a specific document).
+* Docker Compose files orchestrating local deployment of all of these resources.
+
+:::note
+
+Admittedly, the authorization work is incomplete from a feature parity standpoint. And
+the resource authorization via claim sets is going to change in the next milestone. For
+now, you can export from Admin API and re-import that into the DMS Configuration Service
+for some basic resource authorization.
+
+:::
+
+## Ideas
+
+Here are some ideas that might stir your imagination:
+
+* Observability w/streams
+  * Attendance notifications.
+  * Expected record count notifications.
+  * Level 2/3 validation.
+* Stream → Data lake.
+* Stream → ODS-like database.
+* Stream → synchronization to another Ed-Fi API.
+* Machine learning (in OpenSearch?).
+* Serverless execution using cloud functions.
+* Deployment with tools like Kubernetes, OpenShift, TerraForm.
+* Using AWS Aurora or Azure Cosmos DB serverless PostgreSQL.
+* Writing an alternative persistence or query layer (e.g. drop in MongoDB instead of PostgreSQL).
+* UI sitting on top of the Configuration Service.
+* Alternative Open ID Connect provider.
+
+## Next Steps
+
+1. Contact Stephen Fuqua ASAP if you are thinking of developing a "poster".
```

Comment:
Do you want to leave your email here?

Reply:
No :-). Community members whom I want to respond will know how to find me.